### PR TITLE
Add FAIRChem eSEN models

### PR DESCRIPTION
**Pull request by kyonofx**
## Description

Hi Matbench-Disocvery team,

Thank you for maintaining this wonderful community benchmark. We are submitting two checkpoints, a compliant (eSEN-30M-MP) and a non-compliant eSEN models (eSEN-30M-OAM). The eSEN architecture is described in the paper: 

https://arxiv.org/abs/2502.12147

Best,
Xiang

## Checklist

Please check the following items before submitting your PR:

- [x] I have created a new folder and YAML metadata file `models/<arch_name>/<model_variant>.yml` for my submission. `arch_name` is the name of the architecture and `model_variant.yml` includes things like author details, training set names and important hyperparameters.
- [x] I have added the my new model as a new attribute on the [`Model.<arch_name>` enum](https://github.com/janosh/matbench-discovery/blob/57d0d0c8a14cd317/matbench_discovery/enums.py#L274) in `enums.py`.
- [x] I have uploaded the energy/force/stress model prediction file for the WBM test set to Figshare or another cloud storage service (`<yyyy-mm-dd>-<model_variant>-preds.csv.gz`).
- [x] I have uploaded the model-relaxed structures file to Figshare or another cloud storage service (`<yyyy-mm-dd>-wbm-IS2RE-FIRE.json.gz`).
- [x] I have uploaded the phonon predictions to Figshare or another cloud storage service (`<yyyy-mm-dd>-kappa-103-FIRE-<values-of-dist|fmax|symprec>.gz`).
- [x] I have included the urls to the Figshare files in the YAML metadata file (`models/<arch_name>/<model_variant>.yml`). If not using Figshare I have included the urls to the cloud storage service in the description of the PR.
- [x] I have included the test script (`test_<arch_name>_<task>.py` for `task` in `discovery`, `kappa`, `diatomics`) that generated the prediction files.

## Additional Information (Optional)

- [x] I have included a `readme.md` with additional details about my model.

### CODE REVIEW

File: models/eSEN/esen-30m-mp.yml [hunk 1]
```diff
@@ -63,7 +63,7 @@ metrics:
   phonons:
     kappa_103:
       pred_file: models/eSEN/esen-30m-mp/2025-03-17-kappa-103-FIRE-dist=0.03-fmax=1e-4-symprec=1e-5.json.gz
-      pred_file_url: https://figshare.com/files/53054705
```
Comment (CompRhys) [inline models/eSEN/esen-30m-mp.yml]: Can you go onto figshare and make sure the new paths are published publicly?

Comment (kyonofx) [reply models/eSEN/esen-30m-mp.yml]: Hi Rhys, thanks for flagging, just pushed a fix!

Comment (CompRhys) [reply models/eSEN/esen-30m-mp.yml]: still getting insufficient permissions 🙃 if you try "https://figshare.com/files/53054705" in incognito mode do you see the data?

Comment (kyonofx) [reply models/eSEN/esen-30m-mp.yml]: Hi, I previously put a wrong link, this line is replaced in the latest version: 

updated link for eSEN-30M-MP: https://figshare.com/files/53054300
updated link for eSEN-30M-OAM: https://figshare.com/files/53054312

Comment (CompRhys) [reply models/eSEN/esen-30m-mp.yml]: Now the download correctly!

**Comment by CompRhys** [comment]
Thanks for the model submission!

a few more issues with figshare links not being public: 
```
Downloading 'eSEN-30M-MP' from 'https://figshare.com/files/53054669' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-mp/2025-03-17-wbm-IS2RE.csv.gz'
Error downloading url='https://figshare.com/files/53054669'

Downloading 'eSEN-30M-OAM' from 'https://figshare.com/files/53054666' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-oam/2025-03-17-wbm-IS2RE.csv.gz'
Error downloading url='https://figshare.com/files/53054666'
```

but otherwise seemingly SOTA for both compliant and non-compliant. What is the timestep-time/memory-footprint like for this model?

image

**Comment by kyonofx** [comment]
> Thanks for the model submission!
> 
> a few more issues with figshare links not being public (we'll mirror from your figshare to the mbd figshare to ensure they can't be removed once all finalized):
> 
> ```
> Downloading 'eSEN-30M-MP' from 'https://figshare.com/files/53054669' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-mp/2025-03-17-wbm-IS2RE.csv.gz'
> Error downloading url='https://figshare.com/files/53054669'
> 
> Downloading 'eSEN-30M-OAM' from 'https://figshare.com/files/53054666' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-oam/2025-03-17-wbm-IS2RE.csv.gz'
> Error downloading url='https://figshare.com/files/53054666'
> ```
> 
> but otherwise seemingly SOTA for both compliant and non-compliant. What is the timestep-time/memory-footprint like for this model?
> 
> image

Thanks for flagging the figshare issue! I think this is due to me trying to rename the prediction column such that it contains the model name in it.

Both submitted models are of almost the same size so they are very similar in timestep-time/memory-footprint.

Speed: On a single NVIDIA V100 32G GPU, the inference speed is around 4 steps per second (about 350k steps per day) for a system of 71 atoms.
Memory: On a single NVIDIA V100 32G GPU, these models can fit up to ~200 atoms

**Comment by CompRhys** [comment]
> > Thanks for the model submission!
> 
> > 
> 
> > a few more issues with figshare links not being public (we'll mirror from your figshare to the mbd figshare to ensure they can't be removed once all finalized):
> 
> > 
> 
> > ```
> 
> > Downloading 'eSEN-30M-MP' from 'https://figshare.com/files/53054669' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-mp/2025-03-17-wbm-IS2RE.csv.gz'
> 
> > Error downloading url='https://figshare.com/files/53054669'
> 
> > 
> 
> > Downloading 'eSEN-30M-OAM' from 'https://figshare.com/files/53054666' to '/home/runner/work/matbench-discovery/matbench-discovery/models/eSEN/esen-30m-oam/2025-03-17-wbm-IS2RE.csv.gz'
> 
> > Error downloading url='https://figshare.com/files/53054666'
> 
> > ```
> 
> > 
> 
> > but otherwise seemingly SOTA for both compliant and non-compliant. What is the timestep-time/memory-footprint like for this model?
> 
> > 

> Thanks for flagging the figshare issue! I think this is due to me trying to rename the prediction column such that it contains the model name in it. I have now changed the links and hopefully it works.

I think the figshare issue may have been that janosh already mirrored them as was a too many cooks scenario as I didn't spot that.

Thanks for the timings we'd ideally like to capture inference somehow but it's not obvious how to do so fairly.

**Comment by kyonofx** [comment]
> I think the figshare issue may have been that janosh already mirrored them and it was a too many cooks scenario as I didn't spot that.

Yeah that's probably what happened :) Do let me know if there is still issue.

> Thanks for the timings we'd ideally like to capture inference somehow but it's not obvious how to do so fairly.

A benchmark on inference efficiency would be super useful but I agree it is not easy to create a fair benchmark for this. Single and multi-GPU also make a huge difference (e.g., strictly local models may yield a better trade-off for multi-GPU and large systems). The maximize inference speed it would also be helpful to compile the code to C++, all these are important and highly nontrivial.

**Comment by janosh** [comment]
> it was a too many cooks

that was definitely the problem. 😄 sorry, i thought you guys would see the commit message cec0025

@CompRhys i had reverted cd27049 for now as it was breaking `test_data.py` . let's look into that in a separate PR.

@kyonofx i refactored `test_esen_kappa.py` in 2aa5199 to import from the unit-tested `matbench_discovery.phonons`. can you check that it still runs and produces very similar results (not exactly identical since I removed the double relaxation logic)

**Review by janosh — APPROVED** [review]
pending signoff on the `test_esen_kappa.py` changes by @kyonofx but otherwise good to go from my side.

congrats on the excellent performance @kyonofx  👍

**Comment by janosh** [comment]
that's a cool result as it suggests that for the MP model the double (initially constrained) relaxation ends up in a different symmetry than the single relaxation, whereas for the OAM model, it doesn't. perhaps me reading too much into it but suggests that the OAM model's forces are accurate enough not to push atoms off symmetry planes/lines/points.

```txt
eSEN-30M-MP -- before: 0.3213 | after: 0.3398
eSEN-30M-OAM -- before: 0.1710 | after: 0.1704
```

**Comment by kyonofx** [comment]
> that's a cool result as it suggests that for the MP model the double (initially constrained) relaxation ends up in a different symmetry than the single relaxation, whereas for the OAM model, it doesn't.

That's an interesting observation!
